### DISCUSSION TITLE
Expand minio cluster

### DISCUSSION THREAD

**Discussion by sumarnaz** [minio/minio#11728] [Q&A]
Hello.
I'm confused about how to increase the storage capacity of an existing minio cluster.
I have a 4 nodes cluster with 2 drives each.
I started the cluster with the following command on each one of the nodes:
minio server http://minio0{1...4}/mnt/data{1...2}
Everything works as expected.

Now, I would like to increase the total storage. I tried adding a disk to each node and start it with the commands:
minio server http://minio0{1...4}/mnt/data{1...3}
It failed

I then tried to add a new node with 2 disks, as explained it https://docs.minio.io/docs/distributed-minio-quickstart-guide.html, and to start the cluster with the commands:
minio server http://minio0{1...4}/mnt/data{1...2} http://minio05/mnt/data{1...2}
It failed

What am I doing wrong? How can I expand my original 4-nodes x 2-disks cluster?
Thank you.

**Comment by harshavardhana** [comment]
You cannot expand like that, you need to maintain same level of parity SLA.

**Reply by sumarnaz** [reply]
So, the minimum expanded configuration would be to add 4 nodes with 2 drives each?
 http://minio0{1...4}/mnt/data{1...2} http://minio0{5...8}/mnt/data{1...2}

Thanks

**Reply by harshavardhana** [reply]
Yes

**Reply by dtuantran** [reply]
Hello,
Do we need to have the same disk size for the new added pool ?
Thanks.
Regards

**Reply by ravindk89** [reply]
https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html

Each question so far has been covered as part of our docs. Please read through them and let us know if you have further questions, or if a section lacks clarity.

**Reply by dtuantran** [reply]
Thank @ravindk89. I've already read your link and I did research in all documentation. Sorry if I missed some important information. Because in your doc, it doesn't say if new pool must have the same disk size as the original pool.
Let's imagine that we have an original pool of 16 disks 10 TB, could I add new pool of 16 disks of 8 TB ?
Regards,

**Reply by ravindk89** [reply]
As per https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html#minimum-drives-for-erasure-code-parity the only hard requirement is that the number of drives added in the new pool meet the existing parity SLA.

https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html#homogeneous-node-configurations notes that the pool itself should be homogeneous, but it does not have to match any existing pool. That means that the new pool can have larger disks overall.

Thus for your question, the only requirement is meeting EC. If your original pool is 16 drives, then a new pool with 16 drives matches the existing SLAs, so a difference in drive size doesn't matter. You do want all drives to be the same size _within_ the pool.

I can see some areas we can improve the writing and will take down a task for that.

**Reply by dtuantran** [reply]
@ravindk89 Thank you a lot for the explanation ! I asked this question, because I was confused between the "homogeneous nodes" for a pool requirement and for the new pool. We would like to recycle old disks which are all same, but the size is smaller than the existing pool. It's good that we can go ahead.
Thank you again for your help !

**Reply by pternyuk** [reply]
@ravindk89  Thank you for the answer. Can you clarify one more side of this question? Can we meet any performance issues if we use drives with different capacity in different pools? For example is there any difference to use less number of big drives vs more small drives (number is enough to meet EC requirements)

**Reply by zhao6300** [reply]
if I add too many pools, will the read performance be decreased?

**Reply by klauspost** [reply]
@zhao6300 @pternyuk For performance related questions reach out to `sales@min.io` or look at our [subscriptions](https://min.io/pricing).
